system/Config/BaseConfig.php

```diff
@@ -274,21 +274,36 @@ protected function registerProperties()
 
         $shortName = (new ReflectionClass($this))->getShortName();
 
+        if (static::$moduleConfig->registrarHasData) {
```

Comment:
new property on patch PR will cause undefined property unless `isset()` is used.

imo, new option should go to 4.7 branch.